### PR TITLE
Change Cilium templates to standalone version

### PR DESCRIPTION
**Pull request by nebril**
Based on
https://github.com/cilium/cilium/blob/v1.6.0/install/kubernetes/quick-install.yaml

### CODE REVIEW

**Comment by olemarkus** [comment]
This is great! 

It would be good to remove the code that opens the etcd ports to the nodes as well.

**Comment by nebril** [comment]
pls don't merge yet, there is one more problem with the templates

File: k8s/crds/kops_v1alpha2_cluster.yaml [hunk 1]
```diff
@@ -118,10 +122,26 @@ spec:
               properties:
                 aws:
                   properties:
+                    cpuLimit:
+                      description: CPULimit CPU limit of AWS IAM Authenticator container.
```
Comment (chrisz100) [inline k8s/crds/kops_v1alpha2_cluster.yaml]: Why are these limits required for cilium?

File: k8s/crds/kops_v1alpha2_cluster.yaml [hunk 2]
```diff
@@ -654,6 +674,9 @@ spec:
                   items:
                     type: string
                   type: array
+                admissionControlConfigFile:
+                  description: AdmissionControlConfigFile is the location of the admission-control-config-file
```
Comment (chrisz100) [inline k8s/crds/kops_v1alpha2_cluster.yaml]: Is this really required to install cilium?

File: k8s/crds/kops_v1alpha2_cluster.yaml [hunk 3]
```diff
@@ -755,6 +784,19 @@ spec:
...
                   type: string
+                authorizationWebhookCacheAuthorizedTtl:
+                  description: The duration to cache authorized responses from the
+                    webhook token authorizer. Default is 5m. (default 5m0s)
+                  type: string
+                authorizationWebhookCacheUnauthorizedTtl:
+                  description: The duration to cache authorized responses from the
+                    webhook token authorizer. Default is 30s. (default 30s)
+                  type: string
+                authorizationWebhookConfigFile:
```
Comment (chrisz100) [inline k8s/crds/kops_v1alpha2_cluster.yaml]: And once again, looks like you mixed some upstream things into one PR?

File: k8s/crds/kops_v1alpha2_cluster.yaml [hunk 4]
```diff
@@ -2209,29 +2292,53 @@ spec:
...
+                    waitBPFMount:
+                      type: boolean
+                  required:
+                  - enableipv6
+                  - enableipv4
+                  - monitorAggregation
+                  - bpfCTGlobalTCPMax
+                  - bpfCTGlobalAnyMax
+                  - preallocateBPFMaps
+                  - sidecarIstioProxyImage
```
Comment (chrisz100) [inline k8s/crds/kops_v1alpha2_cluster.yaml]: istio for cilium? correct me if I am wrong but they don't belong together I feel?

Comment (nebril) [reply k8s/crds/kops_v1alpha2_cluster.yaml]: Cilium can run alongside istio, docs are here: http://docs.cilium.io/en/v1.6/gettingstarted/istio

**Comment by nebril** [comment]
@chrisz100 I did have some weird changes in this PR, thanks for pointing it out! Should be all good now.

**Comment by chrisz100** [comment]
It’s been long enough for people to also review. 
/approve

**Comment by mikesplain** [comment]
I was planning to review this after all the current CVE's and Centos fixes were in but oh well.

File: pkg/apis/kops/networking.go [hunk 5]
```diff
@@ -155,7 +155,7 @@ type AmazonVPCNetworkingSpec struct {
 	ImageName string `json:"imageName,omitempty"`
 }
 
-const CiliumDefaultVersion = "v1.0-stable"
```
Comment (mikesplain) [inline pkg/apis/kops/networking.go]: Are we really upgrading from 1.0 to 1.6.1?

File: upup/models/cloudup/resources/addons/networking.cilium.io/k8s-1.7.yaml.template [hunk 6]
```diff
@@ -1,412 +1,136 @@
-{{- $etcd_scheme := EtcdScheme }}
-kind: ConfigMap
 apiVersion: v1
```
Comment (mikesplain) [inline upup/models/cloudup/resources/addons/networking.cilium.io/k8s-1.7.yaml.template]: According to https://github.com/kubernetes/kops/pull/7474/files#diff-dec2adb888d136e5fcab51a2af69403dL425 Cilium doesn't support 1.7 so this shouldn't be updated? Also I really don't like forcing old users to this new version, especially when they really shouldn't be using k8s that's this old anyway.

Comment (chrisz100) [reply upup/models/cloudup/resources/addons/networking.cilium.io/k8s-1.7.yaml.template]: I’m a little confused with that statement: people should use sth that old so we shouldn’t force people to update?

File: pkg/apis/kops/networking.go (hunk 5 above)
Comment (chrisz100) [reply pkg/apis/kops/networking.go]: What’s more concerning to me: did we really not care about chili um for 6 minor versions?

File: upup/models/cloudup/resources/addons/networking.cilium.io/k8s-1.7.yaml.template (hunk 6 above)
Comment (mikesplain) [reply upup/models/cloudup/resources/addons/networking.cilium.io/k8s-1.7.yaml.template]: If a user is still running k8s 1.7 - k8s 1.12, forcing a cilium upgrade of this magnitude is likely aggressive and has a strong likelihood of causing instability.

File: pkg/apis/kops/networking.go (hunk 5 above)
Comment (mikesplain) [reply pkg/apis/kops/networking.go]: Anyone is welcome to update at anytime, but especially when leaping versions we need to ensure a smooth upgrade path for all users.

**Comment by mikesplain** [comment]
@nebril Can you test an upgrade from before this PR to post this PR? We might need to break this into smaller upgrades:

According to https://docs.cilium.io/en/v1.5/install/upgrade/#upgrading-from-1-2-x-to-1-3-y

>If you are running Cilium 1.0.x or 1.1.x, please upgrade to 1.2.x first. It is also possible to upgrade from 1.0 or 1.1 directly to 1.3 by combining the upgrade instructions for each minor release. See the documentation for said releases for further information.

**Comment by olemarkus** [comment]
The change with the most impact is probably changing state store from etcd to CRD. There will be network disruptions for the brief time cilium rebuilds the state.

I did a test upgrade from kops 1.13.0 to master now. The add-on version isn't bumped, so the manifests aren't applied. But even with the version bumped, nothing happens. So that is at least one thing to look into. What else needs to happen for the manifests to be applied?

**Comment by olemarkus** [comment]
Cilium resources also miss ```  labels:
    role.kubernetes.io/networking: "1"```

So that should be added to the manifests. But even when I label existing resources and then update, it doesn't change the daemonset.

**Comment by nebril** [comment]
@mikesplain the biggest issue is that Cilium addon was actually broken for several versions since kops closed access to etcd clusters for addons (I don't remember which exact kops version it happened).

I was trying to make a PR with etcd operator, but etcd operator is still hardly production ready, so instead I waited for Cilium to support being backed by CRDs, which landed in 1.6.

**Pull request merged**